homeassistant/components/jellyfin/media_source.py

```diff
@@ -0,0 +1,331 @@
+            result.children = await self._build_albums(artist_id)
+
+        return result
+
+    async def _build_albums(self, artist_id: str) -> list[BrowseMediaSource]:
+        """Return all albums of a single artist as browsable media sources."""
+        albums = await self._get_children(artist_id, ITEM_TYPE_ALBUM)
+        albums = sorted(albums, key=lambda k: k[ITEM_KEY_NAME])  # type: ignore[no-any-return]
+        return [await self._build_album(album, False) for album in albums]
+
+    async def _build_album(
+        self, album: dict[str, Any], include_children: bool
+    ) -> BrowseMediaSource:
+        """Return a single album as a browsable media source."""
+        album_id = album[ITEM_KEY_ID]
+        album_title = album[ITEM_KEY_NAME]
+        thumbnail_url = self._get_thumbnail_url(album)
+
+        result = BrowseMediaSource(
+            domain=DOMAIN,
+            identifier=album_id,
+            media_class=MEDIA_CLASS_ALBUM,
+            media_content_type=MEDIA_TYPE_NONE,
+            title=album_title,
+            can_play=False,
+            can_expand=True,
+            thumbnail=thumbnail_url,
+        )
+
+        if include_children:
+            result.children_media_class = MEDIA_CLASS_TRACK
+            result.children = await self._build_tracks(album_id)
+
+        return result
+
+    async def _build_tracks(self, album_id: str) -> list[BrowseMediaSource]:
+        """Return all tracks of a single album as browsable media sources."""
+        tracks = await self._get_children(album_id, ITEM_TYPE_AUDIO)
+        tracks = sorted(tracks, key=lambda k: k[ITEM_KEY_INDEX_NUMBER])  # type: ignore[no-any-return]
+        return [self._build_track(track) for track in tracks]
+
+    def _build_track(self, track: dict[str, Any]) -> BrowseMediaSource:
+        """Return a single track as a browsable media source."""
+        track_id = track[ITEM_KEY_ID]
+        track_title = track[ITEM_KEY_NAME]
+        mime_type = _media_mime_type(track)
+        thumbnail_url = self._get_thumbnail_url(track)
+
+        result = BrowseMediaSource(
+            domain=DOMAIN,
+            identifier=track_id,
+            media_class=MEDIA_CLASS_TRACK,
+            media_content_type=mime_type,
+            title=track_title,
+            can_play=True,
+            can_expand=False,
+            thumbnail=thumbnail_url,
+        )
+
+        return result
+
+    async def _get_children(
+        self, parent_id: str, item_type: str
+    ) -> list[dict[str, Any]]:
+        """Return all children for the parent_id whose item type is item_type."""
+        params = {
+            "Recursive": "true",
+            "ParentId": parent_id,
+            "IncludeItemTypes": item_type,
+        }
+        if item_type == ITEM_TYPE_AUDIO:
+            params["Fields"] = ITEM_KEY_MEDIA_SOURCES
+
+        result = await self.hass.async_add_executor_job(self.api.user_items, "", params)
```

Comment:
`user_items` seems like a lower level call in the library. It would be better to try to use the higher level API if possible so we don't need to handle details like the params structure in the request.